Set of improvements and refinements to visualizations after React migration

kravets-levko opened the pull request:
## What type of PR is this? (check all applicable)

- [x] Refactor

## Description

- [x] Unify a way we choose background/text colors for visualizations
- [x] Components for visualization editors
  - [x] wrapper for tabbed editors
  - [x] a context help component; open number format specs in drawer instead of new tab;
  - [x] editor section component (https://github.com/getredash/redash/pull/4175#discussion_r329312650)
  - [x] control labels

## Related Tickets & Documents

getredash/redash#3301

kravets-levko commented:
@arikfr @gabrieldutra I touched a lot of files, but I tried to do one change per commit, so you can check commits one by one to see what I did here.

arikfr commented:
All the help links that refer to `redash.io/help`, should just open the HelpDrawer -- no need to have the Popover with the link. For clarity we can add a tooltip saying "Formatting Help".

gabrieldutra commented on client/app/components/visualizations/editor/withControlLabel.jsx:
```diff
@@ -0,0 +1,75 @@
+import './control-label.less';
+
+// TODO: Control label should reflect disabled state
+// <Typography.Text disabled={disabled}>label text</Typography.Text>
```
Perhaps create an issue on GH instead of that? (unless you intend to do it in this PR) I think those comments are more easily lost than the issues :confused:

kravets-levko replied:
I forgot about this :man_facepalming: Now implemented in this PR https://github.com/getredash/redash/pull/4382/commits/7cad47782985c9f49258f77753453bd6c5780755

kravets-levko commented on client/app/components/visualizations/editor/TabbedEditor.jsx:
```diff
@@ -0,0 +1,37 @@
+import { EditorPropTypes } from '@/visualizations';
+
+export default function TabbedEditor({ tabs, options, data, onOptionsChange, ...restProps }) {
+  const optionsChanged = (newOptions, updateFunction = merge) => {
```
You're right - interesting and definitely looks weird. Added some abstraction for it https://github.com/getredash/redash/pull/4382/commits/bd89634ac42cc0ce6dc42b6e7e8e0509d39bffea

gabrieldutra replied:
I tend to like JS simplicity with its weirdness, also assumed the Lodash functions as a "basic set" in the codebase, thus assuming people (most of them at least) will know it or have the docs open. So, for me it was cool anyway.

kravets-levko replied:
I added this little abstraction not because I think somebody may not know how `_.merge` or `_.extend` works, but because somebody may not know how they're used in this particular context. The code code like `onOptionsChange({...}, extend)` raises too much questions: why `extend`? What else could be there? What it extends at all? etc. With something like `onOptionsChange({ ... }, UpdateOptionsStrategy.shallowMerge)` it's easy to understand that second argument defines how first argument will be used to modify options; also, no need to know how `shallowMerge` is internally implemented - you could think about it as a black box, and also you'll know that namespace `UpdateOptionsStrategy` contains other constants that could be used instead of it.

gabrieldutra replied:
I see, considering Visualizations is one of the biggest extension points, it makes sense to leave it as simple and comprehensible as possible. so it's a good abstraction :+1:.

kravets-levko commented:
@arikfr I updated help icon for number formats - now click on it will open help drawer immediately (also added a tooltip):
![image](https://user-images.githubusercontent.com/12139186/69626045-80a5d080-1050-11ea-9112-9376275965f0.png)

Also, I updated help icon for date/time formats to match number format one - it's just a link that opens moment.js help:
![image](https://user-images.githubusercontent.com/12139186/69626248-ef832980-1050-11ea-995a-50bf15ac802a.png)

Other context help blocks are unchanged:
![image](https://user-images.githubusercontent.com/12139186/69626420-4ee13980-1051-11ea-9f11-78c859ccfad8.png)

Also, there are few more places with `redash.io` links (e.g. empty state components) - do you think they should also open a help drawer? If yes - I think it's better to do it in a new PR anyway.

kravets-levko commented:
@arikfr If you're fine with this PR - I'd like to merge it.
